### PR TITLE
📝 Show replies in activity center

### LINKED ISSUE #12239: Show replies in activity center
As a user
I want to see messages that replies to me in the activity center
So that is easier to find the relevant posts

### Descriptions
Replies should be shown in the activity center as per designs

https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=1405%3A345

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #12239 

### Summary

This PR implements reply notifications to activity center. 

#### Screenshots

Captura de Pantalla 2021-07-08 a la(s) 17 28 39

### Testing notes

Reply notifications only work for groups and communities for the time being

### Platforms
- Android
- iOS

### Functional
- activity center

### Steps to test
- Open Status
- Open a group or community chat
- Send a message replying to some user
- Receive notification in activity center

status: ready

### CODE REVIEW

File: src/status_im/ui/screens/notifications_center/styles.cljs [hunk 1]
```diff
@@ -37,6 +42,20 @@
...
+  {:height          22
+   :align-self      :baseline
+   :align-items :center
+   :border-radius   11
+   :border-color    colors/gray-transparent-40
+   :border-width    1
+   :margin-top      6
+   :margin-bottom   10
+   :margin-right    15
+   :padding-left    5
```
Comment (shivekkhurana) [inline src/status_im/ui/screens/notifications_center/styles.cljs]: The odd numbers look odd. Have you looked at `spacing` namespace?

**Review by shivekkhurana — APPROVED** [review]
We can update the styles to use ` quo.design-system.spacing`. Looks good otherwise.

**Review by flexsurfer — APPROVED** [review]

**Comment by qoqobolo** [comment]
@briansztamfater thanks for your work!

### ISSUE 1: Replies in Activity center aren't shown on iOS

**_Steps:_**
1. UserA adds UserB (iOS-user) to contacts and creates a group chat (or community)
2. UserB joins the group, sends a message, and navigates to Home screen
3. UserA sends a reply to UserB
4. UserB opens Activity center
**Expected result:** notification for the reply on Activity center screen
**Actual result:** unread indicator on the bell is shown but Activity center is empty

Also, the unread indicator appears on the bell only for the first received reply. For further replies, only the unread message indicator _on the chat_ is displayed. Sometimes notifications for mentions stop displaying in the activity center as well. To start receiving them again, you need to get a mention while on the Activity Center screen.

https://user-images.githubusercontent.com/67952253/125430054-7f040a8e-ef15-4586-b7b3-6143d14b77b9.mp4

### ISSUE 2: Black font of reply in Activity center
The issue is relevant for several types of messages and is reproducible in develop: https://github.com/status-im/status-react/issues/12274
But if it's not very difficult to fix (and if it's really a regression after [Mention notifications in Activity center](https://github.com/status-im/status-react/pull/12227)), maybe you would like to do it in this PR :)

Screenshot 2021-07-13 at 11 51 51

**Comment by briansztamfater** [comment]
@qoqobolo thanks for the testing, I'm looking at those issues right now.

File: src/status_im/ui/screens/notifications_center/styles.cljs (hunk 1 above)
Comment (briansztamfater) [reply src/status_im/ui/screens/notifications_center/styles.cljs]: I'll take a look at it!

**Comment by briansztamfater** [comment]
@qoqobolo Issues should be addressed, let me know how testing goes :)

**Comment by qoqobolo** [comment]
@briansztamfater thanks, issues 1 and 2 are fixed.

### ISSUE 3: Error and further inability to open Activity center when receiving a specific message in 1-1, group chat, or community

If copy a text from an external website containing an image (e.g. from Wikipedia), which will appear as `obj` when pasted into a message, and send contact request in 1-1 chat/reply/mention in a group or community with this object, then the receiver will get `Error: invalid arity` when trying to open Activity center. 
After that, in most cases, Activity center **cannot be opened**, even after reopening the application. 


**Reproduction**:

**UserA:**
1. Add UserB to contacts
2. Open some website, e.g. wikipedia, and copy some piece of text containing an image, like this:

Screenshot 2021-07-14 at 16 19 25

3. Start 1-1 chat with UserB, paste the copied text (NOT as plain text), and send the message

**UserB:**
1. Open Activity center


#### Actual result
Screenshot 2021-07-14 at 14 41 40

**OS:** Android, iOS
Tried to reproduce in develop, but no success. 

### ISSUE 4: Double `@` when receiving a reply _to a message containing mention of ENS user_

Screenshot 2021-07-14 at 16 40 22

**Reproduction**:
UserA and UserB are members of the same group chat
UserB has ENS

UserA:
- Send a mention of UserB
- Navigate to Activity center

UserB:
- Open the group
- Reply to the mention

UserA:
- Check a new notification in Activity center

**Comment by briansztamfater** [comment]
@qoqobolo Thanks for rising these issues, both should be fixed now!

**Comment by qoqobolo** [comment]
@briansztamfater awesome, thanks!
No new issues were found, failed e2e are not related to PR.
So let's merge!

**Pull request merged**